page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/data/DataProvider.java

```diff
@@ -0,0 +1,57 @@
+package ru.sbtqa.tag.pagefactory.data;
+
+import static java.lang.String.*;
+import ru.sbtqa.tag.datajack.TestDataObject;
+import ru.sbtqa.tag.datajack.adaptors.ExcelDataObjectAdaptor;
+import ru.sbtqa.tag.datajack.adaptors.JsonDataObjectAdaptor;
+import ru.sbtqa.tag.datajack.adaptors.PropertiesDataObjectAdaptor;
+import ru.sbtqa.tag.datajack.exceptions.DataException;
+import ru.sbtqa.tag.qautils.properties.Props;
+
+public class DataProvider {
+
+    private static TestDataObject dataContainer;
+    private static String configCollection;
+
+    public static TestDataObject getInstance() throws DataException {
+        if (dataContainer == null) {
+            configCollection = Props.get("data.initial.collection", null);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define a constant instead of duplicating this literal "data.initial.collection" 4 times. ([squid:S1192](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1192))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=64&issues=AWSn0wiesk71k7lw7bXT&open=AWSn0wiesk71k7lw7bXT)

[ISSUE_KEY]: AWSn0wiesk71k7lw7bXT